sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -332,9 +333,11 @@ private RxDocumentClientImpl(URI serviceEndpoint,
 
         activeClientsCnt.incrementAndGet();
         this.clientId = clientIdGenerator.incrementAndGet();
+        clientMap.put(serviceEndpoint.toString(), clientMap.getOrDefault(serviceEndpoint.toString(), 0) + 1);
```

Comment:
do we need the whole string? or should be just use the `serviceEndpoint.getHost()` - just a thought. 
No problem with the whole string either.

Reply:
I opted for the whole string in case we need port information or something of the sort from the service endpoint - although it could also be shortened by using the authority in order to still get the port. I'm thinking maybe that would be the midpoint between the current and just the host - would this work out?
<img width="199" alt="image" src="https://user-images.githubusercontent.com/30335873/186969494-fcf2604c-956b-4fd1-8329-1a76ae75e504.png">

Reply:
No I guess its okay. I was thinking the overhead of converting the URI `toString` when calling diagnostics serializer. But I think it should be minimal.